exercises/concept/high-score-board/.meta/design.md

```diff
@@ -0,0 +1,94 @@
+- How to create an object literal with `{}` (either empty or with some initial values)
+- What is allowed as keys
+- What is allowed as values (e.g., numbers, strings etc or other objects, arrays or even functions)
+- How to add a key-value pair to an object, how to change the value of an existing key
+- How to retrieve the value (show `obj["key"]` and `obj.key` notation)
+- How to retrieve nested values
+- How to remove an entry
+- How to check a key exists in the object with `hasOwnProperty`
+- How to iterate through the keys with `for ... in`
+
+## Out of Scope
+
+- Prototypes and classes
+- `this`
+- Object destructering
+- null and undefined (will be introduced in another concept exercise, including when they show up in the context of objects, it is a bit hard to tiptoe around this in this concept/exercise but the student can't learn everything at once)
+- `new Object()` / `Object.create`
+- Symbols as keys (should be covered in a separate concept)
+- JSON (should be covered in a separate concept)
+- "pass by reference" (will be introduced in the `functions` concept)
+- optional chaining (will be introduced in the `null-undefined` concept)
+
+## Concepts
+
+The Concepts this exercise unlocks are:
+
+- `objects`
+
+## Prerequisites
+
+- `for-loops` to better understand the for-in loop and because should already have covered some ground in the concept tree before starting with objects (e.g., arrays which are a prerequisite for `for-loops`)
+
+## Analyzer
+
+This exercise could benefit from the following rules in the [analyzer][analyzer]:
+
+1. `createScoreBoard`
+
+   - `essential`: Make sure no class, map etc. was created, there should be just an object.
+   - `actionable`: If the student created an empty object first and then added the value, give feedback to include the entry in the object literal directly.
+   - `actionable`: Check that the object was returned directly, no intermediate assignment to a variable necessary.
+
+2. `addPlayer`
+
+   - ???
+
+3. `removePlayer`
+
+   - `essential`: Make sure `delete` was used and not setting to undefined or null.
+   - `actionable`: If there is additional code to check whether the key is present before deleting it, give feedback that this is not necessary.
+
+4. `updateScore`
+
+   - `actionable`: If the student used a separate variable to calculate the new value first, tell them it is not necessary.
+   - `actionable`: If the student did not use the shorthand assignment operator, tell them about it. If they used it already, give a `celebratory` comment.
+
+5. `applyMondayBonus`
+
+   - `essential`: Check the student actually used for-in.
+   - Same feedback as in `updateScore` applies.
+   - Using `updateScore` in the solution should be treated as equally correct as the exemplar solution.
+
+6. `normalizeScore`
+
+   - `actionable`: No intermediate variables necessary.
+
+## Notes
+
+The exercise is inspired by the [High Score Board Exercise in the Swift track][swift-high-score].
```

Comment:
`forked_from` points to the Julia implementation, and `source` is empty. Perhaps add Elixir to either of those?

I'm not the best person to ask about this tho, @iHiD will likely have a different stance on it.

Reply:
I just saw in the "stories" docs that Elixir is actually the reference implementation. Used that one as forked_from and in the notes now.